File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        int n_cnt = Integer.bitCount(n);
        int next_cnt = 0;
        
        while(true){
            n++;
            next_cnt = Integer.bitCount(n);
            if(n_cnt == next_cnt){
                answer = n;
                break;
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.py
def solution(n):
    answer = 0
    current = bin(n)[2:]
    ocnt = current.count('1')
    # print(ocnt)
    
    nextnum = n + 1
    while True:
        ncnt = bin(nextnum).count('1')
    # print(ncnt)
        if ncnt == ocnt:
            answer = nextnum
            break
        else:
            nextnum += 1
    return answer